### PR TITLE
Add frequently used APIs in the document

### PR DESCRIPTION
**Pull request by annie-xd-wang**
Should we provide additional information for the APIs, or should we list more?

### CODE REVIEW

**Comment by zacsimile** [comment]
This is an awesome start! I think we don't need to add more API calls, but maybe expand on the descriptions. If we could have a bit more info, something analogous to https://micro-manager.org/Version_2.0_API_How_do_I, that would be awesome.

**Comment by zacsimile** [comment]
I like it.

**Review by zacsimile — DISMISSED** [review]

File: docs/source/feature_container/feature_container_home.rst [hunk 1]
```diff
@@ -89,8 +89,131 @@ it has indeed moved) before proceeding.
...
+---------------------------------
+
+Each Feature Object is defined as a class, there is a step-by-step example feature
+that moves the stage to a specific position listed in the multi-position table in the GUI,
+snap an image and calculate its shannon entropy.
+
+1. How to create a custom feature class?
+
+Creating a new feature is the same as a normal new class. However, there is some requires. 
+The first parameter of the ``__init__`` function must be ``model``, which gives the feataure
```
Comment (zacsimile) [inline docs/source/feature_container/feature_container_home.rst]: feature

File: docs/source/feature_container/feature_container_home.rst [hunk 2]
```diff
@@ -89,8 +89,131 @@ it has indeed moved) before proceeding.
...
+Each Feature Object is defined as a class, there is a step-by-step example feature
+that moves the stage to a specific position listed in the multi-position table in the GUI,
+snap an image and calculate its shannon entropy.
+
+1. How to create a custom feature class?
+
+Creating a new feature is the same as a normal new class. However, there is some requires. 
+The first parameter of the ``__init__`` function must be ``model``, which gives the feataure 
+object full access to the model. All the other parameters are keyword arguments and must have default values.
+There should always have a config table.
```
Comment (zacsimile) [inline docs/source/feature_container/feature_container_home.rst]: The feature class should have a `config_table` attribute.

File: docs/source/feature_container/feature_container_home.rst [hunk 3]
```diff
@@ -89,8 +89,131 @@ it has indeed moved) before proceeding.
...
+  
+  from navigate.model.analysis.image_contrast import fast_normalized_dct_shannon_entropy
+  
+  Class FeatureExample:
+
+      def __init__(self, model, position_id=0):
+          self.model = model
+          self.position_id = position_id
+
+          self.config_tabel = {
```
Comment (zacsimile) [inline docs/source/feature_container/feature_container_home.rst]: `config_table`

File: docs/source/feature_container/feature_container_home.rst (hunk 2 above)
Comment (zacsimile) [reply docs/source/feature_container/feature_container_home.rst]: See https://thedeanlab.github.io/navigate/feature_container/feature_container_home.html#the-feature-object for a description of the `config_table`.

File: docs/source/feature_container/feature_container_home.rst [hunk 4]
```diff
@@ -89,8 +89,131 @@ it has indeed moved) before proceeding.
...
+              },
+              "node": {
+                  "device_related": True,
+              }
+          }
+
+2. How to get parameters from GUI?
+
+All the GUI parameters are in the ``model.configuration["experiment"]`` during
+runtime. Here is the example that get positions from the multi-position table
```
Comment (zacsimile) [inline docs/source/feature_container/feature_container_home.rst]: Here is an example that gets positions...

File: docs/source/feature_container/feature_container_home.rst [hunk 5]
```diff
@@ -89,8 +89,131 @@ it has indeed moved) before proceeding.
...
+            self.target_position = positions[self.position_id]
+        else:
+            current_position = self.model.get_stage_position()
+            self.target_position = dict([(axis[:-4], value) for axis, value in current_position.items()])
+
+More GUI parameters can be found in `experiment.yml <https://github.com/TheDeanLab/navigate/blob/develop/src/navigate/config/experiment.yml>`_
+
+3. How to interact with stage?
+
+Now, we move stage to the target position.
```
Comment (zacsimile) [inline docs/source/feature_container/feature_container_home.rst]: Now, we move the stage to the `target_position`.

File: docs/source/feature_container/feature_container_home.rst [hunk 6]
```diff
@@ -89,8 +89,131 @@ it has indeed moved) before proceeding.
...
+.. code-block:: python
+
+    def in_func_signal(self):
+        pos = dict([(f"{axis}_abs", value) for axis, value in self.target_position.items()])
+        self.model.move_stage(pos, wait_until_done=True)
+
+
+4. How to get an image and deal with it?
+
+The image capatured by the camara will be stored in the data buffer.
```
Comment (zacsimile) [inline docs/source/feature_container/feature_container_home.rst]: captured by the camera

File: docs/source/feature_container/feature_container_home.rst [hunk 7]
```diff
@@ -89,8 +89,131 @@ it has indeed moved) before proceeding.
...
+    def in_func_signal(self):
+        pos = dict([(f"{axis}_abs", value) for axis, value in self.target_position.items()])
+        self.model.move_stage(pos, wait_until_done=True)
+
+
+4. How to get an image and deal with it?
+
+The image capatured by the camara will be stored in the data buffer. 
+You should add your code to deal with the image in the main data function.
+Here is an example to calculate the shannon entropy of the image.
```
Comment (zacsimile) [inline docs/source/feature_container/feature_container_home.rst]: Here is an example that calculates the Shannon entropy of the image.

File: docs/source/feature_container/feature_container_home.rst [hunk 8]
```diff
@@ -89,8 +89,131 @@ it has indeed moved) before proceeding.
...
+.. code-block:: python
+
+      def in_func_data(self, frame_ids):
+          for id in frame_ids:
+              image = self.model.data_buffer[id]
+              entropy = fast_normalized_dct_shannon_entropy(image,
+                  psf_support_diameter_xy=3)
+              print("entropy of image:", id, entropy)
+
+Now, you've create a whole new feature and can use it as your wish.
```
Comment (zacsimile) [inline docs/source/feature_container/feature_container_home.rst]: as you wish

File: docs/source/feature_container/feature_container_home.rst [hunk 9]
```diff
@@ -89,8 +89,131 @@ it has indeed moved) before proceeding.
...
+Here is an example to open shutter:
+
+.. code-block:: python
+
+  self.model.active_microscope.shutter.open_shutter()
+
+
+* How to pause and resume data threads in the model
+
+The image data acquired from the camara are handle in an indepandant thread.
```
Comment (zacsimile) [inline docs/source/feature_container/feature_container_home.rst]: are handled

File: docs/source/feature_container/feature_container_home.rst [hunk 10]
```diff
@@ -89,8 +89,131 @@ it has indeed moved) before proceeding.
...
+Some devices need more time to get ready or move, then you may need to pause
```
Comment (zacsimile) [inline docs/source/feature_container/feature_container_home.rst]: Some devices need more time to get ready or move. In this case, you may need to pause

File: docs/source/feature_container/feature_container_home.rst [hunk 11]
```diff
@@ -89,8 +89,131 @@ it has indeed moved) before proceeding.
...
+
+- `prepare_next_channel() <../../html/_autosummary/navigate.model.microscope.Microscope.html#navigate.model.microscope.Microscope.prepare_next_channel>`_
+- `move_stage() <../../html/_autosummary/navigate.model.microscope.Microscope.html#navigate.model.microscope.Microscope.move_stage>`_
+- `get_stage_position() <../../html/_autosummary/navigate.model.microscope.Microscope.html#navigate.model.microscope.Microscope.get_stage_position>`_
+- `update_stage_limits() <../../html/_autosummary/navigate.model.microscope.Microscope.html#navigate.model.microscope.Microscope.update_stage_limits>`_
+- `move_remote_focus() <../../html/_autosummary/navigate.model.microscope.Microscope.html#navigate.model.microscope.Microscope.move_remote_focus>`_
+
+Please visit `Model APIs <../../html/_autosummary/navigate.model.model.Model.html#navigate.model.microscope.Microscope.calculate_all_waveform>`_ and `Microscope APIs <../../html/_autosummary/navigate.model.microscope.Microscope.html>`_ for more details.
+
+Creating Custom Feature List
```
Comment (zacsimile) [inline docs/source/feature_container/feature_container_home.rst]: Creating a Custom Feature List

File: docs/source/feature_container/feature_container_home.rst (hunk 1 above)
Comment (annie-xd-wang) [reply docs/source/feature_container/feature_container_home.rst]: I've updated the document. Thank you.

**Review by zacsimile — APPROVED** [review]

**Pull request merged**